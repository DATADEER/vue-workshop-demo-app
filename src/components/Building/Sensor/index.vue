<template>
	<div>
		<h3>{{ room_name }}</h3>
		<div>IP: {{ ip_address }}</div>
		<div>Mac: {{ mac_address }}</div>
		<div>Sensor: {{ temperature }}Â° C</div>
	</div>
</template>

<script>
export default {
	props: {
		room_name: String,
		ip_address: String,
		mac_address: String,
		temperature: Number
	}
};
</script>
