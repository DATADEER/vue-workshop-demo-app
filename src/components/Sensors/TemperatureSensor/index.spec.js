import Sensor from ".";
import { shallowMount } from "@vue/test-utils";

const mockedProps = {
  room_name: "Bedroom",
  ip_address: "127.0.0.1",
  mac_address: "00:0d:83:b1:c0:8e",
  temperature: 22
};

describe("Sensor", () => {
  it("renders without crashing", () => {
    expect(() => shallowMount(Sensor)).not.toThrow();
  });

  it("renders props", () => {
    const wrapper = shallowMount(Sensor, { propsData: mockedProps });
    expect(wrapper).toMatchSnapshot();
  });

  describe("renders prop", () => {
    const wrapper = shallowMount(Sensor, { propsData: mockedProps });

    it("room_name", () => {
      expect(wrapper.find("h3").text()).toContain(mockedProps.room_name);
    });

    it("ip_address", () => {
      expect(
        wrapper
          .findAll("div")
          .at(1)
          .text()
      ).toContain(mockedProps.ip_address);
    });

    it("mac_address", () => {
      expect(
        wrapper
          .findAll("div")
          .at(2)
          .text()
      ).toContain(mockedProps.mac_address);
    });

    it("temperature", () => {
      expect(
        wrapper
          .findAll("div")
          .at(3)
          .text()
      ).toContain(mockedProps.temperature);
    });
  });
});
