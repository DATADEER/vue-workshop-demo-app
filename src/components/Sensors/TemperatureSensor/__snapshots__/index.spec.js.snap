// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sensor renders props 1`] = `
<div>
  <h3>Bedroom</h3>
  <div>IP: 127.0.0.1</div>
  <div>Mac: 00:0d:83:b1:c0:8e</div>
  <div>Sensor: 22Â° C</div>
</div>
`;
